test/lib/tasks/generateLibraryManifest.js

```diff
@@ -0,0 +1,484 @@
+const {test} = require("ava");
+
+const ui5Builder = require("../../../");
+const tasks = ui5Builder.builder.tasks;
+const path = require("path");
+const ui5Fs = require("@ui5/fs");
+const resourceFactory = ui5Fs.resourceFactory;
+
+function createWorkspace() {
+	return resourceFactory.createAdapter({
+		virBasePath: "/",
+		project: {
+			metadata: {
+				name: "test.lib"
+			},
+			version: "2.0.0",
+			dependencies: [
+				{
+					metadata: {
+						name: "sap.ui.core"
+					},
+					version: "1.0.0"
+				}
+			]
+		}
+	});
+}
+
+function createDependencies() {
+	return resourceFactory.createAdapter({
+		fsBasePath: path.join(__dirname, "..", "..", "fixtures", "sap.ui.core-evo", "main", "src"),
+		virBasePath: "/resources"
+	});
+}
+
+async function assertCreatedManifest(t, oExpectedManifest) {
+	const {workspace, dependencies, resources} = t.context;
+
+	await Promise.all(resources.map((resource) => workspace.write(resource)));
```

Comment:
Just out of curiosity: why is writing needed although the resources have been created already? To make globbing work?

Reply:
Yes. The resources are created without the context of the workspace. So by just creating them, they can't be found in the task.
When the processor would be tested standalone, then the plain resources could be handed.

When writing the test I noticed that this isn't really straight forward so I think we could improve the fs API to be more elegant.